crypto/src/swap.rs

```diff
@@ -109,14 +368,38 @@ impl TryFrom<&[u8]> for SwapCiphertext {
     }
 }
 
-// TODO: ideally this would live in `component/dex/` or maybe a new subcrate including only
-// dex-related logic, but the former causes cyclic import issues
 #[derive(Debug, Clone)]
 pub struct TradingPair {
     pub asset_1: AssetId,
     pub asset_2: AssetId,
 }
 
+impl TradingPair {
+    /// Convert the trading pair to bytes.
+    pub fn to_bytes(&self) -> [u8; 64] {
+        let mut result: [u8; 64] = [0; 64];
+        result[0..32].copy_from_slice(&self.asset_1.0.to_bytes());
+        result[32..64].copy_from_slice(&self.asset_2.0.to_bytes());
+        result
+    }
```

Comment:
ditto above in re: encapsulating fixed-size serialization, do we want this to be public?